### ISSUE TITLE
create a tool to make it very easy to make snapshots of examples

### ISSUE THREAD

**Issue by josharian** [cursorless-dev/cursorless#1744]
[split out from https://github.com/cursorless-dev/cursorless/pull/1538]

 A common need both when writing documentation and interacting via slack is the ability to run a command and show its results. This is complicated by the need to show hats,  cursor state, selection state, etc. We currently resort to some amount of ascii art to do this, which is easy, but suboptimal in a number of ways.

What if we wrote a tool that automated all of this? You'd start with a document like:

```markdown
# bring air to bat

big art
```

and trigger the example generator and it would spit out an image like

![demo](https://github.com/cursorless-dev/cursorless/assets/67496/6908b8ee-0542-4dfd-a448-5e710658bacb)

but maybe less ugly? I manually created that. (it would figure out the content range, look for the header line, run the command in the header line, take screenshots before and after, and stitch them altogether.)

https://github.com/kufii/CodeSnap/ is an extension that lets you take snapshots of code. I'm not sure whether it handles hats, but we could probably adapt the underlying code to do so. (Seems like the core is https://github.com/tsayen/dom-to-image.) You'd have to figure out how to handle the blinking cursor. :/

**Comment by josharian** [comment]
cc @jmegner

**Comment by josharian** [comment]
Actually, it would probably be better not to put the spoken command into the screenshot. And instead capture it, and render it to text with the default spoken form, and then composite that above the rest of the image. Anyway, details TBD--the interesting thing is the idea here. :)

**Comment by pokey** [comment]
Yeah this might be a nice intermediate version of https://github.com/cursorless-dev/cursorless/pull/940

**Comment by pokey** [comment]
I think https://github.com/cursorless-dev/cursorless/issues/215 is also related

**Comment by pokey** [comment]
And see also https://github.com/cursorless-dev/cursorless/issues/866

**Comment by josharian** [comment]
it definitely seems like an idea that keeps coming up. :) it seems like the next step might be to think through which of these flavors is worth working on first.
